src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -471,7 +471,7 @@ protected function getIndirectObject($pdfData, $xref, $obj_ref, $offset = 0, $de
         }
         $objref = $obj[0].' '.$obj[1].' obj';
         // ignore leading zeros
-        $offset += strspn($pdfData, '0', $offset);
+        $offset += strspn($pdfData, "0\n", $offset);
```

Comment:
Could you please describe how you come up with this fix?

Reply:
Yes, this is more like a dirty fix. The offset given by `RawDataParser::getXrefData` is off by a newline for the given PDF which does not fit assumptions in `RawDataParser::getIndirectObject`. But i am new to this package and not so familiar with PDF-standards, so i was afraid of changing regex in `getXrefData`. To me all seamed to work fine when adding newlines to the trimming in `getIndirectObject`. But i will review `RawDataParser::getXrefData` again, so maybe wait a bit if you intend to merge.

Reply:
Ok I did not find anything on cross-ref offsets and leading newlines. So i guess it is fine to throw it away. Assuming "Microsoft: Print To PDF" has correctly implemented PDF-1.7-standard & given PDF was not corrupted afterwards.

Reply:
I think in the tc-lib-pdf-parser project this was solved by the `$offset++` there
See https://github.com/tecnickcom/tc-lib-pdf-parser/pull/6/files

Reply:
Ok i tried this solution, works as well. Not sure which one is better.

Reply:
### About TCPDF
There are so many repositories from tecnickcom for PDF parser: 
* **https://github.com/tecnickcom/TCPDF**
* https://github.com/tecnickcom/tc-lib-pdf-parser
* https://github.com/tecnickcom/tc-lib-pdf

Previously used in this library was **TCPDF**. But i removed it and moved some of its code to us (Ref: #299).

---

### About this fix

First of all, its great that you support us here, so please don't get me wrong. But i am not sure, if you know **why** your changes fix the issue. You wrote it yourself before: `Yes, this is more like a dirty fix.`. 

You added `\r\n` to include line breaks, which is Windows specific. Others like Linux use `\n` for instance. That leads to the question if your fix also covers PDFs which were generated on a Mac for instance?

To be honest, i still lack detailed knowledge about PDF parsing. So I can't decide if your fix works or not. Furthermore I just started to get used to the code of this library too. **That said I really would like to hear a third opinion on that change.**

This library is used by >600 projects (said Github) and they relying on us to keep it stable. If you can point me to some specification (or something like that) which supports the change you made, that would help.

Reply:
Hey @k00ni , i totally understand that. I am not expecting this to be merged without further review.

To my understanding the cross reference table is parsed just fine in TCPDF. 

_The byte offset in the decoded stream shall be a 10-digit number, padded with leading zeros if necessary,
giving the number of bytes from the beginning of the file to the beginning of the object._ 
(https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/PDF32000_2008.pdf)

but i did not find good resources if an object can be padded at that offset? which microsoft seems to be doing?
so i think a solution in `RawDataParser::getIndirectObject` might be the right choice. ("this is a dirty fix" was my understanding yesterday)

i referenced the tcpdf commit because it seems to address a similar issue. this was just fyi. and it does basically the same.

I dont get your point on the line breaks? i am removing both (CR & LF) which should work for windows and unix, am i wrong?

Reply:
fyi: Setasign/FPDI seems to do this for \x20\x0A\x0C\x0D\x09\x00

see PdfParser/Tokenizer.php

Reply:
or https://github.com/mozilla/pdf.js/blob/v2.5.207/src/core/obj.js#L1504

Reply:
as i understand https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDFReference13.pdf:

_PDF treats any sequence of consecutive white-space characters as if there were just one_

_For the most part, EOL markers are treated the same as any other white-space characters._

and **3.2.9 Indirect Objects** defines no special meaning for EOL.